[12.0][MIG] stock_cycle_count: Migration to 12.0

bodedra opened the pull request:
Depends on:
------------------

- [x] https://github.com/OCA/stock-logistics-warehouse/pull/622 stock_inventory_discrepancy
- [x] https://github.com/OCA/stock-logistics-warehouse/pull/623 stock_inventory_exclude_sublocation

LoisRForgeFlow committed:
[ADD] stock_cycle_count

MiquelRForgeFlow committed:
[ADD] Adding tests

LoisRForgeFlow committed:
[IMP] stock_cycle_count:
 * Update README.
 * Fixes: DEFAULT_SERVER_DATETIME_FORMAT, PERCENT variables and sale price calculation.

LoisRForgeFlow committed:
[IMP] tests.

LoisRForgeFlow committed:
[9.0][FIX] Solves issue when doing picking with several moves.

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: add the possibility to define rules for specific zones.

LoisRForgeFlow committed:
[9.0][FIX] stock_cycle_count: Do not propose cycle counts in the past.

LoisRForgeFlow committed:
[9.0][FIX] stock_cycle_count: Fix a bug when combining automatic and manual creation of cycle counts.

LoisRForgeFlow committed:
[9.0][FIX] stock_cycle_count: fields only editable in draft and add tracking to some of them.

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: Enhance tree view with filter and colors.

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: The configuration of an inventory adjustment related to a cycle count cannot be modified.

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: update existing cycle counts instead of cancel them.

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: add server action to confirm several CC at a time

oca-transbot committed:
OCA Transbot updated translations from Transifex

MiquelRForgeFlow committed:
[FIX] stock_cycle_count (usage cannot be 'view')

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[10.0][MIG] stock_cycle_count

LoisRForgeFlow committed:
cycle_count: fix cycle count state issue

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: fix accuracy computation and store it

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: basic_accuracy report

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: enhance views

LoisRForgeFlow committed:
[9.0][IMP] stock_cycle_count: cycle counts report

oca-transbot committed:
OCA Transbot updated translations from Transifex

LoisRForgeFlow committed:
[10.0][IMP] fix api issues and clean up code

LoisRForgeFlow committed:
[10.0] stock_cycle_count: adapt to follow OCA convention.

LoisRForgeFlow committed:
[10.0][IMP] stock_cycle_count:
* simplify fetching latest inventory date
* use api.depends where needed
* add hook for cycle count creation
* take advantage of api.multi on check_zero_confirmation method

LoisRForgeFlow committed:
[10.0][UPD] stock_cycle_count: add roadmap.

oca-travis committed:
[UPD] Update stock_cycle_count.pot

LoisRForgeFlow committed:
[11.0][MIG] stock_cycle_count

JordiBForgeFlow committed:
stock_cycle_count: fix issue with nan while doing the mean calculation

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update stock_cycle_count.pot

oca-transbot committed:
Update translation files

Updated by Update PO files to match POT (msgmerge) hook in Weblate.

pedrobaeza committed:
[FIX] stock_cycle_count: Don't lose return value

For avoiding an error validating pickings.

Commit:
[12.0][MIG] stock_cycle_count: Migration to 12.0

bodedra commented:
@jbeficent @lreficent @aheficent 

Would you please help me find out root cause to fail `('Zero confirmation not being created.')` test case ?

![Screenshot_2019-06-25 Job #3890 3 - OCA stock-logistics-warehouse - Travis CI](https://user-images.githubusercontent.com/14229503/60082931-6c347380-9752-11e9-82c9-3e9e4e7e203f.png)

LoisRForgeFlow commented:
@bodedra just opened a PR to fix the remaining issue: https://github.com/ursais/stock-logistics-warehouse/pull/2. Could you review?

bodedra commented:
@lreficent 

![480b6216b16fb1a21acc318c3d805f53](https://user-images.githubusercontent.com/14229503/60730426-b7c0fb80-9f62-11e9-8462-d24cd40de6a3.jpg)

Thanks for your fix.

Now all test case are passed. Still ci/runbot fail.

MiquelRForgeFlow commented on stock_cycle_count/reports/report_stock_location_accuracy.py:
```diff
@@ -0,0 +1,37 @@
+
+
+class LocationAccuracyReport(models.AbstractModel):
+    _name = "report.stock_location_accuracy"
```
```suggestion
    _name = "report.stock_location_accuracy"
    _description = "Location Accuracy Report"
```

bodedra replied:
Done. @mreficent

rousseldenis commented on stock_cycle_count/models/stock_warehouse.py:
```diff
@@ -40,8 +39,7 @@ def get_horizon_date(self):
             self, parent):
-        domain = [('parent_left', '>=', parent.parent_left),
-                  ('parent_right', '<=', parent.parent_right),
+        domain = [('parent_path', '=like', parent.parent_path + '%'),
```
I'm wondering why child of/parent of would not works here ?

See  https://github.com/odoo/odoo/blob/12.0/odoo/osv/expression.py#L755

bodedra replied:
@rousseldenis  I use it as a reference of https://github.com/odoo/odoo/blob/12.0/addons/stock/models/product.py#L206 and https://github.com/odoo/odoo/blob/12.0/addons/stock/models/product.py#L211

rousseldenis replied:
Yeah, that's weird because it's in the same commit that implements parent_path to optimize queries...

bodedra replied:
Additionally, parent_left and parent_right fields are removed from stock.location object.

rousseldenis replied:
Yes, that was the aim of parent_path: to optimize child of/parent of search operators, so, normally, you should use them as before.

bodedra replied:
so what do you recommend on these ?

rousseldenis reviewed (approved):
Code review

rousseldenis commented:
@bodedra Could you squash your commits ?

bodedra commented:
@rousseldenis Sure.

bodedra commented:
@rousseldenis @mreficent Thanks for your review and suggestion.

@lreficent can you update your review ?

LoisRForgeFlow committed:
[12.0][FIX/MIG] stock_cycle_count: fix zero confirmation rules.
Now zero-qty quants are not removed straight away but by a cron job.
Therefore, we have to also check that the quants are not zero-qty ones.

LoisRForgeFlow reviewed (approved):
LGTM (just fixed history a little to recover my contribution to the migration)

Let's merge after travis goes green.

LoisRForgeFlow commented:
/ocabot merge
